Cookbook/Technical-Documents/CRP-Bot-ImplementationDetails.md

```diff
@@ -0,0 +1,108 @@
+
+Any section that is not part of the `knownProjects` list (static constant declared in `configure.swift`) will be skipped during this process. Even though tickets on unknown boards will be included in the CRP ticket's body, there will be no attempt to create a JIRA version on any unknown board nor touch the Fix Version field of those tickets. 
+
+This whitelist has two main purposes:
+
+1. First, to avoid accidentally creating release and touch tickets that are not supposed to be our responsibility and might have been wrongfully detected when parsing the commit messages. For example if a commit message contains a reference to the original ticket but also mentions another ticket from a board we don't own, we don't want to process the latter one and affect a board and ticket we don't own
+2. Some boards, even if they are supposed to follow the CRP process, are misconfigured in JIRA and use a custom set of fields and custom process instead of using the same fields and configuration than all the other boards. 
+
+Those misconfigured boards are tracked in [IOSP-101](https://babylonpartners.atlassian.net/browse/IOSP-101). 
+
+Trying to execute the CRP automation on those boards would end in an invalid API request or worse, some unexpected changes in the tickets, because field IDs would not match the ones expected by the bot.
+
+### `batchSetFixedVersions`
+
+This method is called for each board by `createAndSetFixedVersions`. It is passed the list of tickets of said board, and the JIRA version reference in order to update the `Fix Version` field for these tickets.
+
+It mainly consists of applying `map` on all the tickets, making an API call to `setFixVersion(version, for: ticket, …)` on each ticket in turn.
+
+### FixedVersionReport
+
+One particularity of `createAndSetFixedVersions` and `batchSetFixedVersions` is that we don't want any API call failure to interrupt the whole process. Instead, in case of any API failure, we gather the error in a report object and continue to the next ticket.
+
+This report is then printed at the very end of the CRP process (as a reply in Slack) to inform the Release Engineer which tickets failed to have their "Fix Version" field updated, to let them update the tickets manually.
+
+Note: The `FixedVersionReport` struct is used to gather messages. It's implemented such as passing multiple `FixedVersionReport` to its `init` will `flatMap` the messages of them all in a single report, allowing us to use constructs like `map(to: FixedVersionReport.self, on: container, FixedVersionReport.init)` on the parallel sequence of API calls, thus setting the version on each ticket in `batchSetFixedVersions`. 
+
+A similar construct is used in `createAndSetFixedVersions` to gather all the reports on each board/`ChangelogSection` into a single report.
+
+### SlowClient
+
+One of the particularities of this process is that it sends a **LOT** of API calls to JIRA:
```

Comment:
```suggestion
One of the particularities of this process is that it sends a **LOT** of API calls to JIRA:

```

(Need empty line between paragraph and list for markdown rendering to work properly)